components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/service/impl/ActionManagementServiceImpl.java

```diff
@@ -404,4 +407,16 @@ private Action buildAction(String actionType, ActionDTO actionDTO) {
                 .rule(actionDTO.getActionRule())
                 .build();
     }
+
+
+    /**
+     * Build ActionDTO from Action object for audit logging purposes.
+     *
+     * @param action The Action object containing all fields including timestamps from database.
+     * @return ActionDTO object with timestamp fields populated for audit logging.
+     */
+    private ActionDTO buildAuditActionDTO(Action action) {
```

Comment:
Lets not have a private method specially for this case.

Reply:
I think this is not correct. Building the actionDTO with this will not catch the action properties. 
Can you check this again ?